src/systemTest/java/com/flightstats/hub/testcase/WebhookErrorTest.java

```diff
@@ -0,0 +1,136 @@
+package com.flightstats.hub.testcase;
+
+import com.flightstats.hub.model.Webhook;
+import lombok.SneakyThrows;
+import lombok.extern.slf4j.Slf4j;
+import org.junit.Before;
+import org.junit.Test;
+import org.slf4j.Logger;
+
+import java.util.Optional;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+@Slf4j
+public class WebhookErrorTest extends WebhookTest {
```

Comment:
I have a pretty staunchly held belief that inheritance is almost always the wrong way to go when structuring code. This is the subclass of a subclass, so I feel like the inheritance is getting too deep (and I have a suggestion in another PR for getting rid of BaseTest as a base class). I'd split WebhookTest into a few classes (webhooks, callbacks, channel, maybe one for shared behavior) and instantiate them here.

Reply:
Yeah, I think the changes in Manjula's k8s branch will require significant re-writes to this, so I didn't spend much time on the design here.

Reply:
Fair enough. I'll watch for the refactoring in another PR.

Reply:
I agree with Laurie here. Multi level inheritance can make things complex and is error prone. I would consider adding helper classes here.

Reply:
Yep, I agree.  I can refactor this if you'd like, it just seemed not worth investing much in since it's going to change so much once your branch comes in.

Reply:
Don't worry about my branch. Go head. I will deal with it.